src/main/kotlin/com/virtuslab/dokka/site/processors.kt

```diff
@@ -180,23 +185,53 @@ class SitePagesCreator(cxt: DokkaContext) : BaseStaticSiteProcessor(cxt) {
             val children =
                 if (noChildren) emptyList() else from.listFiles()?.mapNotNull { renderDocs(it) } ?: emptyList()
             if (children.count { it.isIndexPage } > 1)
-                println("ERROR: Multiple index pages found ${children.filter { it.isIndexPage }}") // TODO proper error handling
+                println("ERROR: Multiple index pages found ${children.filter { it.isIndexPage }}") // TODO (#14): provide proper error handling
 
-            val templateFile = if (from.isDirectory) null else loadTemplateFile(from)
-            val content = try {
+            val templateFile = loadTemplateFile(
+                if (from.isDirectory) {
+                    val indexFiles = from.listFiles { file -> file.name == "index.md" || file.name == "index.html" }
+                    check(indexFiles!!.size == 1) { "ERROR: Multiple index pages found under ${from.path}" } // ensured above too
+                    indexFiles.first()
+                } else from
+            )
+
+
+            val resolvedPage = try {
                 val context = RenderingContext(emptyMap(), layouts)
                 if (from.isDirectory) {
                     children.find { it.isIndexPage }?.resolved ?: EmptyResolvedPage
-                } else templateFile!!.resolve(context)
+                } else templateFile.resolve(context)
             } catch (e: Throwable) {
                 val msg = "Error rendering $from: ${e.message}"
-                println("ERROR: $msg") // TODO proper error handling
+                println("ERROR: $msg") // TODO (#14): provide proper error handling
                 ResolvedPage(msg, emptyList())
             }
 
-            DocPageNode(from, templateFile, children.filter { !it.isIndexPage }, content, dri)
+            val parser =
+                if (templateFile.isHtml) HtmlParser()
```

Comment:
This is ok for now, but I do not think we should parse HTML that way in long run (I will create issue for it)